lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -55,6 +55,13 @@ def _setup(
         self.error_payload = error_payload
         self.empty_es_response_payload = empty_es_response_payload
         self.index_from_metadata = index_from_metadata
+        if API_METHOD.GET in self.cls_obj.schemas:
+            self.api_method = API_METHOD.GET
+        elif API_METHOD.POST in self.cls_obj.schemas:
+            self.api_method = API_METHOD.POST
+        else:
+            self.api_method = None
+            raise Exception("Unexpected API method")
```

Comment:
Hmm. My first thought was that you don't really need to set `self.api_method` here since you're aborting the constructor with an exception. Except ... in this weird pytest hackery, it's *not* really a constructor. In any case, I don't think that setting `self.api_method` to `None` is really going to help much if the test continues, so that's probably not the best solution.

Another option is to `assert` that the class has either `API_METHOD.GET` or `API_METHOD.POST` so that the test fails immediately. _Or_ else we could `pytest.skip` because the common test class isn't applicable.